### PR TITLE
Refactor our protocol interfaces into `@0x/contracts-interfaces`

### PR DESCRIPTION
**Pull request by LogvinovLeon**
## Description

Moves protocol interfaces into a separate package so that other packages including those interfaces but being used by the protocol package (extensions) don't create a dependency cycle.

## Testing instructions

Tests still pass.

## Types of changes

* New feature (non-breaking change which adds functionality)

* Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [x] Prefix PR title with `[WIP]` if necessary.
*   [x] Add tests to cover changes as needed.
*   [x] Update documentation as needed.
*   [x] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Comment by abandeali1** [comment]
Not sure I like this change. Why can't extensions just import from the interfaces directory to get around circular dependencies? This also seems like it's less of an issue if the extensions are in their own package.

**Comment by LogvinovLeon** [comment]
@abandeali1 Extensions can't live in the same package as Exchange contract because they need to be versioned separately. Extensions depend on Exchange interfaces. But in order to test that functionality in Exchange, we use those Extensions as example implementations of those interfaces.

**Comment by abandeali1** [comment]
I see, so the issue is basically with the `transactions.ts` tests. Would it solve the problem if we just moved the tests that use any extensions/examples into relevant packages? If there is any way to remove the circular dependencies that doesn't involve moving the interfaces to a separate package, that would be my preference.

**Comment by abandeali1** [comment]
I see, so the issue is basically with the `transactions.ts` tests. Would it solve the problem if we just moved the tests that use any extensions/examples into relevant packages? If there is any way to remove the circular dependencies that doesn't involve moving the interfaces to a separate package, that would be my preference.

**Comment by abandeali1** [comment]
I see, so the issue is basically with the `transactions.ts` tests. Would it solve the problem if we just moved the tests that use any extensions/examples into relevant packages? If there is any way to remove the circular dependencies that doesn't involve moving the interfaces to a separate package, that would be my preference.

File: contracts/interfaces/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+## Contracts interfaces
```
Comment (fabioberger) [inline contracts/interfaces/README.md]:
```suggestion
## Contract interfaces
```

File: contracts/interfaces/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Smart contract interfaces of the 0x protocol.
+
+## Usage
+
+Contracts that make up and interact with version 2.0.0 of the protocol can be found in the [contracts](./contracts) directory. The contents of this directory are broken down into the following subdirectories:
```
Comment (fabioberger) [inline contracts/interfaces/README.md]: Think we can remove the mention of `2.0.0` specifically in all these packages now.

File: contracts/interfaces/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+*   [protocol](./contracts/protocol)
+    *   This directory contains the contract interfaces that make up version 2.0.0. A full specification can be found [here](https://github.com/0xProject/0x-protocol-specification/blob/master/v2/v2-specification.md).
+
+## Bug bounty
+
+A bug bounty for the 2.0.0 contracts is ongoing! Instructions can be found [here](https://0xproject.com/wiki#Bug-Bounty).
```
Comment (fabioberger) [inline contracts/interfaces/README.md]: Is this still true?

Comment (LogvinovLeon) [reply contracts/interfaces/README.md]: Yes

**Pull request merged**